src/main/java/teammates/common/datatransfer/NotificationTargetUser.java

```diff
@@ -0,0 +1,68 @@
+package teammates.common.datatransfer;
+
+import teammates.common.util.Const;
+
+/**
+ * Represents the type of target user groups for notifications.
+ */
+public enum NotificationTargetUser {
+
+    /**
+     * Target users are students only.
+     */
+    STUDENT(Const.NotificationTargetUser.STUDENT),
+
+    /**
+     * Target users are instructors only.
+     */
+    INSTRUCTOR(Const.NotificationTargetUser.INSTRUCTOR),
+
+    /**
+     * Target users are both instructors and students.
+     */
+    GENERAL(Const.NotificationTargetUser.GENERAL);
+
+    private final String targetUser;
+
+    NotificationTargetUser(String targetUser) {
+        this.targetUser = targetUser;
+    }
+
+    public String getTargetUser() {
+        return targetUser;
+    }
+
+    /**
+     * Formats the participant type as a singular noun.
+     *
+     * @return A user-friendly {@code String} representing this participant in singular form.
+     */
+    public String toSingularFormString() {
+        switch (this) {
+        case STUDENT:
+            return "student";
+        case INSTRUCTOR:
+            return "instructor";
+        case GENERAL:
+            return "general";
+        default:
```

Comment:
Is a default case necessary for an enum? Assumed it should fit into one of the enums specified.

Reply:
Without the default case, there is a chance that the method does not return anything from the syntax. So a default statement is needed.